src/SignalR/common/Http.Connections/src/ConnectionEndpointRouteBuilderExtensions.cs

```diff
@@ -48,20 +48,14 @@ public static class ConnectionEndpointRouteBuilderExtensions
         public static IEndpointConventionBuilder MapConnectionHandler<TConnectionHandler>(this IEndpointRouteBuilder endpoints, string pattern, Action<HttpConnectionDispatcherOptions> configureOptions) where TConnectionHandler : ConnectionHandler
         {
             var options = new HttpConnectionDispatcherOptions();
-            // REVIEW: WE should consider removing this and instead just relying on the
-            // AuthorizationMiddleware
-            var attributes = typeof(TConnectionHandler).GetCustomAttributes(inherit: true);
-            foreach (var attribute in attributes.OfType<AuthorizeAttribute>())
-            {
-                options.AuthorizationData.Add(attribute);
-            }
             configureOptions?.Invoke(options);
```

Comment:
In MVC we ended up combining the `AuthorizeAttribute` decorated on the controller with metadata on endpoint routing: https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/Authorization/AuthorizeFilter.cs#L148-L152

The concern was that you could have done something like so:

```C#
[Authorize("Some-Claim")]
public class UserController
{

}

...
mvcOptions.Filters.Add(new AuthorizeFilter(new AuthorizeAttribute("Different-Claim")));
```

And you may have a auth policy that only makes sense when it observes both `Some-Claim` and `Different-Claim` together. In the SignalR case, you kinda get in to a similar place if the user configures `HttpConnectionDispatcherOptions`:

```C#
[Authorize("Some-Claim")]
public class MyHub { ..}

MapHub(..., options => options.AuthorizeData.Add(new AuthorizeAttribute("Different"));
```

We have a couple of options here:

1) Copy `HttpConnectionDispatcherOptions.AuthorizeData` to EndpointMetadata. The auth will run for the full set once and then again for the values specified in `HttpConnectionDispatcherOptions.AuthorizeData`

2) Add annotated attributes to `HttpConnectionDispatcherOptions.AuthorizeData` if the user configured it

3) Throw if the user configured  `HttpConnectionDispatcherOptions.AuthorizeData` when using endpoint routing and direct them to use endpoint routing's extension methods to configure auth.

Reply:
For 1, couldn't we copy the data over and then replace the options with an empty list so it only runs once?

Reply:
Sure, that's an option. As long the user doesn't get to look at the collection later on.

Reply:
Yeah the issue is that all the attributes play an important role determining which schemes to authenticate, so it potentially has a big impact on whether authorization succeeds

Reply:
This should be fixed now, turns out we were already copying the data over to EndPoints but now I'm clearing the list so we don't run Auth again.